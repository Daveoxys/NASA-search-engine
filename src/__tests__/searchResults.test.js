import { render, screen } from "@testing-library/react";
import SearchResults from "../components/searchResults";

describe("SearchResults", () => {
  const validProps = ["moon", "earth", "stars"];
  test("It renders images", () => {
    const { asFragment } = render(<SearchResults results={validProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("It renders the correct amount of images if the search exists", () => {
    render(<SearchResults results={validProps} />);
    const images = screen.getAllByAltText("spaceImage");

    expect(images.length).toEqual(3);
    expect(images[0]).toBeInTheDocument();
    expect(images[0]).toHaveClass("search-results__image");
  });
});
