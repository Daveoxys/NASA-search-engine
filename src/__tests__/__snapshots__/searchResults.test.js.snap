// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults It renders images 1`] = `
<DocumentFragment>
  <div
    class="search-results"
  >
    <img
      alt="spaceImage"
      class="search-results__image"
      src="moon"
    />
    <img
      alt="spaceImage"
      class="search-results__image"
      src="earth"
    />
    <img
      alt="spaceImage"
      class="search-results__image"
      src="stars"
    />
  </div>
</DocumentFragment>
`;
